src/array.c

```diff
@@ -27,11 +27,13 @@ struct array_s {
     size_t   size;
     size_t   capacity;
     float    exp_factor;
+    bool     is_sorted;
     void   **buffer;
 
     void *(*mem_alloc)  (size_t size);
     void *(*mem_calloc) (size_t blocks, size_t size);
     void  (*mem_free)   (void *block);
+    int   (*comp)       (const void *a, const void *b);
```

Comment:
Suggest renaming this to 'cmp'. 'comp' may refer to 'compression' while 'cmp' is a common abbreviation for 'compare'.